### PR TITLE
Kernel: More misc bare metal fixes

### PR DESCRIPTION
**Pull request by supercomputer7**
The goal is to aid us in bare metal development.
Therefore, I'll add a way to add an embedded kernel commandline. Also, adding history for `VirtualConsole` would be nice so I decided to work on it too.

### CODE REVIEW

File: Kernel/CommandLine.cpp [hunk 1]
```diff
@@ -59,6 +56,20 @@ UNMAP_AFTER_INIT CommandLine::CommandLine(const String& string)
...
 }
 
+UNMAP_AFTER_INIT CommandLine::CommandLine(const String& string)
+{
+    s_the = this;
+    auto embedded_cmdline = StringView(s_embedded_cmd_line);
+    StringBuilder builder;
+    builder.append(embedded_cmdline);
+    builder.append(" ");
+    builder.append(string.substring_view(0));
```
Comment (IdanHo) [inline Kernel/CommandLine.cpp:66]: any reason for not just doing: `builder.append(string);` ?

Comment (supercomputer7) [reply Kernel/CommandLine.cpp:66]: Not really, now that I see it works OK with `String` too :)

File: Kernel/CommandLine.cpp [hunk 2]
```diff
@@ -4,13 +4,15 @@
...
 
+#include <AK/StringBuilder.h>
 #include <Kernel/CommandLine.h>
 #include <Kernel/Panic.h>
 #include <Kernel/StdLib.h>
 
 namespace Kernel {
 
 static char s_cmd_line[1024];
+static const char* s_embedded_cmd_line = "";
```
Comment (linusg) [inline Kernel/CommandLine.cpp:15]: Can we make this

```suggestion
static constexpr StringView s_embedded_cmd_line = "";
```

? Then we shouldn't need the StringView construction in `build_commandline()`.

File: Kernel/CommandLine.cpp [hunk 3]
```diff
@@ -34,15 +36,24 @@ UNMAP_AFTER_INIT void CommandLine::initialize()
 {
     VERIFY(!s_the);
     s_the = new CommandLine(s_cmd_line);
+    dmesgln("Kernel Commandline: {}", kernel_command_line().string());
 }
 
-UNMAP_AFTER_INIT CommandLine::CommandLine(const String& string)
-    : m_string(string)
+UNMAP_AFTER_INIT void CommandLine::build_commandline(String cmdline_from_bootloader)
```
Comment (linusg) [inline Kernel/CommandLine.cpp:42]: Can we take a const reference here? or `move()` the string below?

File: Kernel/CommandLine.cpp [hunk 4]
```diff
@@ -34,15 +36,24 @@ UNMAP_AFTER_INIT void CommandLine::initialize()
...
 {
-    s_the = this;
+    auto embedded_cmdline = StringView(s_embedded_cmd_line);
+    StringBuilder builder;
+    builder.clear();
```
Comment (linusg) [inline Kernel/CommandLine.cpp:46]: This should not be needed

File: Kernel/CommandLine.cpp [hunk 5]
```diff
@@ -34,15 +36,24 @@ UNMAP_AFTER_INIT void CommandLine::initialize()
...
+    builder.append(cmdline_from_bootloader);
+    if (!embedded_cmdline.is_null() && !embedded_cmdline.is_empty()) {
```
Comment (linusg) [inline Kernel/CommandLine.cpp:48]: `!is_empty()` implies `!is_null()`, so

```suggestion
    if (!embedded_cmdline.is_empty()) {
```

**Comment by supercomputer7** [comment]
Let's wait a bit with `VirtualConsole` history, because I want to make sure I'm making it so it will fit in LibVT code as well.

**Pull request merged**